drivers/can/can_stm32fd.c

```diff
@@ -55,8 +56,14 @@ static int can_stm32fd_get_core_clock(const struct device *dev, uint32_t *rate)
 
 static void can_stm32fd_clock_enable(void)
 {
+#ifdef CONFIG_SOC_SERIES_STM32U5X
+	LL_RCC_PLL1_EnableDomain_48M();
+	LL_APB1_GRP2_EnableClock(LL_APB1_GRP2_PERIPH_FDCAN1);
+	LL_RCC_SetFDCANClockSource(LL_RCC_FDCAN_CLKSOURCE_PLL1);
+#else
```

Comment:
Since you are ignoring the `CONFIG_CAN_STM32FD_CLOCK_SOURCE` setting if `CONFIG_SOC_SERIES_STM32U5X=y` you need to reflect that dependency in `Kconfig.stm32fd`.

Reply:
Another option could be to add a new choice to `CONFIG_CAN_STM32FD_CLOCK_SOURCE` and make that choice depend on `CONFIG_SOC_SERIES_STM32U5X=y`.

Reply:
Good catch! Kconfig is definitely something I'm still getting used to working with.

I've tried adding a new choice with a dependency as shown below, but for whatever reason the Kconfig is evaluated as missing the dependency even if I set the configuration directly above the statement
```
config CAN_STM32FD_CLOCK_SOURCE_PLL1Q
	bool "PLL1Q"
	depends on CONFIG_SOC_SERIES_STM32U5X
	help
	  PLL1 "Q" clock used as FDCAN clock source.
```
```
// In prj.conf
CONFIG_SOC_SERIES_STM32U5X=y
CONFIG_CAN_STM32FD_CLOCK_SOURCE_PLL1Q=y
```
I've also provided them in the board `*_defconfig` file, and that doesn't seem to take either. Instead, it's just providing the default `CONFIG_CAN_STM32FD_CLOCK_SOURCE_HSE` for the Kconfig.

Any thoughts?

Reply:
You need to drop the `CONFIG_` prefix when referencing the config within Kconfig files:

```
config CAN_STM32FD_CLOCK_SOURCE_PLL1Q
	bool "PLL1Q"
	depends on SOC_SERIES_STM32U5X
	help
	  PLL1 "Q" clock used as FDCAN clock source.
```